backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/SessionListeningJettyFactory.java

```diff
@@ -0,0 +1,69 @@
+package org.molgenis.emx2.web;
+
+import javax.servlet.http.HttpSessionListener;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.util.thread.QueuedThreadPool;
+import org.eclipse.jetty.util.thread.ThreadPool;
+import spark.ExceptionMapper;
+import spark.embeddedserver.EmbeddedServer;
+import spark.embeddedserver.EmbeddedServerFactory;
+import spark.embeddedserver.jetty.EmbeddedJettyServer;
+import spark.embeddedserver.jetty.JettyHandler;
+import spark.embeddedserver.jetty.JettyServerFactory;
+import spark.http.matching.MatcherFilter;
+import spark.route.Routes;
+import spark.staticfiles.StaticFilesConfiguration;
+
+// copied from EmbeddedJettyFactory
+public class SessionListeningJettyFactory implements EmbeddedServerFactory {
```

Comment:
Maybe add a little comment here that the class is used to add the session listener, since the servlet context is inaccessible for embedded jetty.